src/ServiceControl/Infrastructure/DomainEvents/DomainEventBusPublisher.cs

```diff
@@ -0,0 +1,22 @@
+namespace ServiceControl.Infrastructure.DomainEvents
+{
+    using NServiceBus;
+
+    class DomainEventBusPublisher : IDomainHandler<IDomainEvent>
```

Comment:
Not sure if needwd but if polymorphic dispatch would be available we could handle here IEvent

Reply:
Currently the `DomainEvents` class does not handle polymorphic dispatch. It only matches the exact type and the root type (`IDomainEvent`). I guess we could make it more smart. Want to try? ;-) We can PP

Reply:
I'm ok with that